Add release notes for 7.1.0-preview (#864)
Azure/AppConfiguration: releaseNotes/MicrosoftExtensionsConfigurationAzureAppConfiguration.md

# Microsoft.Extensions.Configuration.AzureAppConfiguration
[Source code][source_code] | [Package (NuGet)][package]

## 7.1.0-preview - January 19, 2024
### Enhancements

> * The provider will now automatically discover other existing replicas and attempt to connect to them when the AppConfig store and any replicas passed by the user have failed. This is enabled by default and can be disabled by setting the new `AzureAppConfigurationOptions.ReplicaDiscoveryEnabled` property equal to `false`. [#442](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/442)

Reviewer: ```suggestion
* The provider will now automatically discover replicas and use them in fail-over scenarios. This is enabled by default and can be disabled by setting the new `AzureAppConfigurationOptions.ReplicaDiscoveryEnabled` property equal to `false`. [#442](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/442)
```

I don't think the ordering bit is too important.

Reviewer: This is so important I might even suggest it to be it's own section.
 
Like

### Replica auto discovery

The provider now supports automatic discovery of replicas. ...

Author: Yea I think that's good point, I could create a new section and add a more detailed summary like what @zhenlan suggested if you agree with that.

Reviewer: Yes, sounds good.

> * The provider will now automatically discover other existing replicas and attempt to connect to them when the AppConfig store and any replicas passed by the user have failed. This is enabled by default and can be disabled by setting the new `AzureAppConfigurationOptions.ReplicaDiscoveryEnabled` property equal to `false`. [#442](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/442)

Reviewer 2: We shouldn't use "AppConfig" in official documentation. The "store and replicas" doesn't sound right. Replicas are part of a store. Suggest below:

For App Configuration stores with geo-replication enabled, the provider will automatically discover any additional replicas and attempt to connect to them when it fails to connect to user-provided endpoints in code. This capability allows applications to leverage geo-replication for enhanced resiliency without code change and redeployment.

Author: I like the summary, should I not mention the property?

Reviewer 2: I skipped mentioning it because yours looks good. Sorry for the confusion. Yes, we should mention it. It's a new API.

* Added support for the `variants`, `allocation`, and `telemetry` properties for feature flags introduced in version 4.0.0-preview of the [Microsoft.FeatureManagement](https://github.com/microsoft/FeatureManagement-Dotnet/releases/tag/4.0.0-preview) library. [#476](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/476)

### Bug Fixes
* Fixed a bug where passing a null value for the `key` parameter to `AzureAppConfigurationRefreshOptions.Register` would throw a `NullReferenceException`. [#503](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/503)

> * Fixed a bug with `AzureAppConfigurationOptions.Select`, `FeatureFlagOptions.Select`, and `AzureAppConfigurationOptions.SelectSnapshot` where additional calls with the same filters after the first call were ignored. The provider will now correctly use only the last call with duplicate filters to preserve label precedence. [#490](https://github.com/Azure/AppConfiguration-DotnetProvider/issues/490)

Reviewer 2: > ... preserve label precedence.

Is it just label, not key and label?

Author: I was trying to describe the scenario where multiple select calls have the same key but different label, and the label used is prioritized based on the order. I could say "key and label" if that makes more sense?

Reviewer 2: ... or we simply say "... to preserve the expected precedence"?

# Microsoft.Extensions.Configuration.AzureAppConfiguration
[Source code][source_code] | [Package (NuGet)][package]

## 7.1.0-preview - January 19, 2024
### Replica Auto Discovery

> * For App Configuration stores with geo-replication enabled, the provider will now automatically discover any additional replicas and attempt to connect to them when it fails to connect to user-provided endpoints in code. This capability allows applications to leverage geo-replication for enhanced resiliency without code change and redeployment. Replica discovery is enabled by default and can be disabled by setting the new `AzureAppConfigurationOptions.ReplicaDiscoveryEnabled` property equal to `false`. [#442](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/442).

Reviewer 2: I understand we want to emphasize this but having a separate section is a bit confusing what it is about. Does it make sense to nest it under enhancements but have a bolded "Replica auto discovery" at the beginning?

Author: It makes sense to have under enhancements, and yea that seems to be the best way to add a title to a bullet point that I can find. The other thought I had was to make the `Enhancements` section say `Other Enhancements` instead.

### Enhancements

> * The provider will now automatically discover other existing replicas and attempt to connect to them when the App Configuration store and any replicas passed by the user have failed. This is enabled by default and can be disabled by setting the new `AzureAppConfigurationOptions.ReplicaDiscoveryEnabled` property equal to `false`. 

Reviewer 2: Should this be deleted?

Author: Yes, accidentally left it
